src/util/address.rs

```diff
@@ -256,9 +240,12 @@ impl Address {
     /// This address type was introduced with BIP16 and is the popular type to implement multi-sig these days.
     #[inline]
     pub fn p2sh(script: &script::Script, network: Network) -> Address {
+        let mut hash_engine = ScriptHash::engine();
+        hash_engine.input(&script[..]);
+
         Address {
             network: network,
-            payload: Payload::ScriptHash(ScriptHash::hash(&script[..])),
```

Comment:
Well... In this situation, the script slice already exists, so there is no advantage of using the engine...